torch/csrc/jit/codegen/cuda/test/test_gpu_transpose.cpp

```diff
@@ -789,6 +790,60 @@ TEST_F(NVFuserTest, FusionViewNoTranspose_CUDA) {
   TORCH_CHECK(!hasAtLeastTwoValidGroups(&fusion));
 }
 
+TEST_F(NVFuserTest, FusionTransposeSelfMapping_CUDA) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = transpose(tv0, 0, 1);
+  auto tv2 = add(tv0, tv1);
+  fusion.addOutput(tv2);
+
+  EXPECT_THAT(
+      [&]() { IterDomainGraph(fusion_ptr.get()); },
+      testing::ThrowsMessage<c10::Error>(testing::HasSubstr("not supported")));
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn({5, 5}, options);
+
+  FusionExecutorCache executor_cache(std::move(fusion_ptr));
+  auto cg_outputs = executor_cache.runFusionWithInputs({t0});
+
+  auto ref = t0.transpose(0, 1) + t0;
+
+  testValidate(
+      executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
+}
+
+#if 0
+// silent wrong result
+TEST_F(NVFuserTest, FusionTransposeViewSelfMapping_CUDA) {
```

Comment:
cc @csarofeen

Reply:
This doesn't get caught?

Reply:
Random thought: we might want to make sure the disjoint view sets, and something similar for axes involved in permutation match. The issue here seems to be that we're transposing the same partial dimensions. We probably want to make sure throughout the fusion the dimensions active in a transpose are dimensions that are disjoint in the view set.

Reply:
Yeah, thinking through the analysis, I don't know if we'd need to build something like a transpose disjoint map, but I think it might be enough if we check all the dimensions active in a transpose are disjoint. I'm wondering if we could somehow get in trouble with a series of transposes that are all individually disjoint in the view map, but could exhibit behavior above where we're effectively transposing "partial" dimensions.

Reply:
Any thoughts on this @zasdfgbnm ?

Reply:
> might be enough if we check all the dimensions active in a transpose are disjoint

I feel that this is in the wrong direction. For example `torch.zeros(6).view(2, 3).transpose(0, 1)` should be totally fine to schedule, but the transpose dimensions are not disjoint in the view disjoint map.

In my opinion, using solely disjoint-set is not sufficient for modeling this problem. That is, we can not just come up with a new fancy disjoint-set of IterDomains such that a fusion is schedulable iif its disjoint-set is really disjoint.

I think we should do the following analysis: For the DAG
```
       T0[I0, I1]
       /            \
   view   transpose
      |               \
T1[I2, I3]   T2[I4, I5]
           \       /
        T3[I6, I7]
```
we first start from the view operation. What the view operation tells us is:
1. On the producer side, {I0, I1} must be ordered as (I0, I1) and I0 is contiguous.
2. On the consumer side, {I2, I3} must be ordered as (I2, I3) and I2 is contiguous.
If we propagate this order and contiguity information, along the DAG, at some point, we will see a conflict. For example, if we propagate T0->T2 and T1->T3->T2, then:
1. from the first path, we will conclude that {I4, I5} must be ordered as {I5, I4} and I5 is contiguous.
2. from the second path, we will conclude that {I4, and I5} must be ordered as {I4, I5} and I4 is contiguous.
The above 1 and 2 are in conflict, so we reject this fusion.

Reply:
Should we merge this PR, and discuss and fix the skipped test in a follow-up? Currently, transpose is already enabled on TorchScript in TOT devel, and this would prevent the bug from happening on transpose.

Reply:
Yeah, let's merge this, then please move your comment to an issue and we can discuss further there.